component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/voiceotp/VoiceOTPUtils.java

```diff
@@ -0,0 +1,561 @@
+            }
+        } catch (UserStoreException e) {
+            throw new VoiceOTPException("Failed while trying to access userRealm of the user : " + username, e);
+        }
+        return false;
+    }
+
+    /**
+     * Update the mobile number (user attribute) in user's profile.
+     *
+     * @param username  the Username
+     * @param attribute the Attribute
+     * @throws VoiceOTPException
+     * @throws UserStoreException
+     */
+    public static void updateUserAttribute(String username, Map<String, String> attribute, String tenantDomain)
+            throws VoiceOTPException, UserStoreException {
+
+        try {
+            // updating user attributes is independent from tenant association.not tenant association check needed here.
+            UserRealm userRealm;
+            // user is always in the super tenant.
+            userRealm = VoiceOTPUtils.getUserRealm(tenantDomain);
+            if (userRealm == null) {
+                throw new VoiceOTPException("The specified tenant domain " + tenantDomain + " does not exist.");
+            }
+            // check whether user already exists in the system.
+            VoiceOTPUtils.verifyUserExists(username, tenantDomain);
+            UserStoreManager userStoreManager = userRealm.getUserStoreManager();
+            userStoreManager.setUserClaimValues(username, attribute, null);
+        } catch (AuthenticationFailedException e) {
+            throw new VoiceOTPException("Exception occurred while connecting " +
+                    "to User Store: Authentication is failed. ", e);
+        }
+    }
+
+    /**
+     * Verify whether user Exist in the user store or not.
+     *
+     * @param username the Username
+     * @throws VoiceOTPException
+     */
+    public static void verifyUserExists(String username, String tenantDomain) throws VoiceOTPException,
+            AuthenticationFailedException {
+
+        UserRealm userRealm;
+        boolean isUserExist = false;
+        try {
+            userRealm = VoiceOTPUtils.getUserRealm(tenantDomain);
+            if (userRealm == null) {
+                throw new VoiceOTPException("Super tenant realm not loaded.");
+            }
+            UserStoreManager userStoreManager = userRealm.getUserStoreManager();
+            if (userStoreManager.isExistingUser(username)) {
+                isUserExist = true;
+            }
+        } catch (UserStoreException e) {
+            throw new VoiceOTPException("Error while validating the user.", e);
+        }
+        if (!isUserExist) {
+
+            log.debug("User does not exist in the User Store");
+
+            throw new VoiceOTPException("User does not exist in the User Store.");
+        }
+    }
+
+    /**
+     * Get the user realm of the logged in user.
+     *
+     * @param tenantDomain the tenantDomain
+     * @return th user realm
+     * @throws AuthenticationFailedException
+     */
```

Comment:
Let's format the method comments properly in all instances to increase readability.